File: go/find-target-indices-after-sorting-array.go
func targetIndices(nums []int, target int) []int {
    to_left := 0
    num_targets:= 0

    for i:=0; i < len(nums); i++{
        if nums[i] == target {
            num_targets++
        }else if nums[i] < target{
            to_left++
        }
    }    

    ans := []int{}
    for i:= to_left; i < num_targets+to_left; i++{
        ans = append(ans, i)
    }

    return ans
}

File: rust/find-target-indices-after-sorting-array.rs
impl Solution {
    pub fn target_indices(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut num_left = 0;
        let mut targets = 0;

        for i in 0..nums.len(){
            if nums[i] < target {
                num_left += 1;
            }else if nums[i] == target {
                targets+=1;
            }
        }

        let mut ans: Vec<i32> =  Vec::new();

        for i in num_left..(num_left+targets){
            ans.push(i);
        }  

        ans
    }
}
